develop-docs/sdk/telemetry/spans/batch-processor.mdx

```diff
@@ -95,3 +94,93 @@ Scenario: 1 span added application crashes
+
+With this option, the BatchProcessor first stores its data in a thread-safe FIFO queue, residing in an async-safe memory space, allowing the crash reporter to write to disk when a crash occurs. Furthermore, the BatchProcessor stores data asynchronously into a file, allowing it to recover after an abnormal termination, for which the crash handler can't run.
+
+The BatchProcessor maintains its logic of batching multiple logs and spans together into a single envelope to avoid multiple HTTP requests.
+
+Hybrid SDKs pass every log and span down to the native SDKs, which will put every log and span in their BatchProcessor and its cache when logs and spans are ready for sending, meaning after they go through beforeLog, integrations, processors, etc.
+
+#### Receiving Data
+
+When the BatchProcessor receives data, it performs the following steps
+
+1. Put the item into the FIFO queue on the calling thread.
+2. On a background thread, serialize the next item of the FIFO queue and store it in the `cache` file.
+3. Remove the log from the FIFO queue.
+4. If the queue isn't empty, go back to step 2.
+
+The FIFO queue has a `max-item-count` of 64. When the FIFO queue exceeds `max-item-count`, the BatchProcessor MUST drop items and record client reports with the category `queue_overflow` for every dropped item. SDKs MAY choose a different `max-item-count` value, if needed. SDKs MUST NOT expose the `max-item-count` value to users as an option. We can make this option public in the future, if required.
+
+#### Abnormal Process Termination
+
+When SDKs detect an abnormal process termination, they MUST write the items in the FIFO queue to the `abnormal-termination-x` file where `x` is the an increasing index of the file starting from 0.
+
+When the process terminates abnormally and the SDKs can't detect it, the SDKs lose items in the FIFO queue, which we accept over blocking the calling thread that could be the main thread.
+
+No matter the abnormal process termination, SDKs MUST send items both in the `abnormal-termination-x` and `cache` files on the next SDK start. Please refer to the [SDK start](#sdk-start) section for the detailed specification.
+
+#### Cache File Location
+
+The SDK SHOULD store the BatchProcessor cache files in a folder that is a sibling of the `envelopes` or `replay` folder, named `batch-processor`. This folder is scoped per DSN, so SDKs ensure not mixing up data for different DSNs. The `batch-processor` folder MAY contain the follow files, where `x` is the an increasing index of the file starting from 0:
+
+- `cache` - The active writing cache file for the BatchProcessor.
+- `cache-to-flush-x` - The cache file that the BatchProcessor is about to convert to an envelope.
```

Comment:
we call it `flushing` above, I guess we should be consistent with naming?

Reply:
This is not relevant anymore as I removed this section.